tests/unit_tests/scheduler/test_job.py

```diff
@@ -0,0 +1,228 @@
+import asyncio
+import json
+import shutil
+from typing import List
+from unittest.mock import AsyncMock, MagicMock
+
+import pytest
+
+import ert
+from ert.ensemble_evaluator._builder._realization import Realization
+from ert.load_status import LoadResult, LoadStatus
+from ert.run_arg import RunArg
+from ert.scheduler import Scheduler
+from ert.scheduler.driver import JobEvent
+from ert.scheduler.job import STATE_TO_LEGACY, Job, State
+
+
+def create_scheduler():
+    sch = MagicMock()
+    sch._events = asyncio.Queue()
+    sch.driver = AsyncMock()
+    return sch
+
+
+@pytest.fixture
+def realization():
+    run_arg = RunArg(
+        run_id="",
+        ensemble_storage=MagicMock(),
+        iens=0,
+        itr=0,
+        runpath="test_runpath",
+        job_name="test_job",
+    )
+    realization = Realization(
+        iens=run_arg.iens,
+        forward_models=[],
+        active=True,
+        max_runtime=None,
+        run_arg=run_arg,
+        num_cpu=1,
+        job_script=str(shutil.which("job_dispatch.py")),
+    )
+    return realization
+
+
+async def assert_scheduler_events(
+    scheduler: Scheduler, job_events: List[JobEvent]
+) -> None:
+    await asyncio.sleep(0)
```

Comment:
await asyncio.sleep(0) at start and end of this function should not give anything extra. Since it is an async function, it must be surrounded by await in the calling scope, which accomplished the same yielding.

Reply:
I am slowing starting to kinda get the hang of async, i think :)